test/tests/experiment.js

```diff
@@ -0,0 +1,105 @@
+/* @flow */
+
+import { experiment } from '../../src/experiment';
+
+describe('experiment', () => {
+    const name = 'potatoLicker';
+    it('should call logComplete function that returns the result of experiment function call', () => {
+        const expObj = experiment({ name });
+        const logCompleteResult = expObj.logComplete({});
+        if (expObj !== logCompleteResult) {
+            throw new Error(`Expected calling logComplete function to equal expObject returned from experiment function call`);
+        }
+    });
+    it('should call logStart function that returns the result of experiment function call', () => {
+        const expObj = experiment({ name });
+        const logStartResult = expObj.logStart({});
+        if (expObj !== logStartResult) {
+            throw new Error(`Expected calling logStart function to equal expObject returned from experiment function call`);
+        }
+    });
+    it('should call logStart function and NOT call logTreatment function when localStorage is not set', () => {
+        let isCalled;
+        const logTreatment = () => {
+            isCalled = true;
+        };
+        const expObj = experiment({ name, logTreatment });
+        expObj.logStart();
+        expObj.log(name);
+        if (isCalled) {
+            throw new Error(`Expected logTreatment function to not have been called`);
+        }
+    });
+    it('should call logStart function and call logTreatment function when localStorage is set', () => {
+        let isCalled;
+        const logTreatment = () => {
+            isCalled = true;
+        };
+        window.localStorage.setItem('__belter_experiment_storage__', JSON.stringify({
```

Comment:
Is it possible to make this test work without writing to `__belter_experiment_storage__`? This is intended to be an implementation detail, so it may cause the test to break if it ever changes in future.

Reply:
I could not figure out a way to make this test work without writing to `__belter_experiment_storage__`.
The reason I had to set a value to that localStorage is because the `isEventUnique` function can return either true or false. `isEventUnique` will return `true` without giving  `__belter_experiment_storage__` the correct value, so I changed the value of `__belter_experiment_storage__` in order for it to return `false`.
Here is the line of code that this test covers:
https://github.com/krakenjs/belter/blob/master/src/experiment.js#L19

Reply:
But this test is checking if `logTreatment` is called, right? It is called in the case that the event *is* unique. So it should be called even if there's no pre-existing value in localStorage right?

Reply:
Ah you are right. I checked the functions again and realize that the test description should be `when localStorage is NOT set` rather than `is set`

Reply:
Yep, it looks like I can't pass this test without writing to `__belter_experiment_storage__`.
Interestingly, I also tried to do `window.localStorage.setItem('__belter_experiment_storage__', '');` and that also made me pass the test -- so it looks like the criteria of that localstorage being set is a way to get `logTreatment` to be called.

Ah, you are right -- this test isn't the one that covers [line 19](https://github.com/krakenjs/belter/blob/master/src/experiment.js#L19). The one that covers line 19 is on line 21 of this test file (`the test right before this one`).

It seems this test is actually already covered by another test in this file too (line 14 of this test file), so I can remove this test if that is best. I thought I had to cover the case of when `isEventUnique` returns true(by seeing if logtreatment gets called), but it looks like that line 14 of this test file already does that.

Reply:
In place of this test, I can do this:
```
 it('should call logStart function that returns the result of experiment function call', () => {
        let isCalled;
        const logTreatment = () => {
            isCalled = true;
        };
        const expObj = experiment({ name, logTreatment });
        expObj.logStart({});
        if (isCalled) {
            throw new Error(`Expected logTreatment function to not have been called`);
        }
    });
```
Even though line 14 of this test file covers the case of when `isEventUnique` return true, that test isn't made for that purpose. So I think it may be better to add this additional test to be more explicit.

Thoughts?

Reply:
> Interestingly, I also tried to do window.localStorage.setItem('__belter_experiment_storage__', ''); and that also made me pass the test

Aha! This tells us the problem :) The test is being affected by `localStorage` state from a previous test.

We should avoid this by doing this to reset the state before each test run:

```javascript
beforeEach(() => {
    window.sessionStorage.clear();
    window.localStorage.clear();
});
```